test/extended/dr/machine_recover.go

```diff
@@ -77,123 +73,69 @@ var _ = g.Describe("[sig-cluster-lifecycle][Feature:DisasterRecovery][Disruptive
 					Resource: "machines",
 				}).Namespace("openshift-machine-api")
 
-				// framework.Logf("Verify etcd endpoints are healthy")
-				// certsDir := "/tmp/etcd-client-certs/"
-				// dumpEtcdCertsOnDisk(oc, certsDir)
-				// defer os.RemoveAll(certsDir)
-				// unhealthyEtcds := getUnhealthyEtcds(certsDir, masters)
-				// o.Expect(len(unhealthyEtcds)).To(o.BeNumerically("==", 0))
-
-				// createMachineHealthCheckForRole("master")
-				// defer deleteMachineCheckForRole("master")
 				createMachineHealthCheckForRole("worker")
 				defer deleteMachineCheckForRole("worker")
 
 				replacedWorkerMachineName := getMachineNameByNodeName(oc, replacedWorker.Name)
-				// replacedMasterMachineName := getMachineNameByNodeName(oc, replacedMaster.Name)
-				// replacedMasterMachine, err := ms.Get(context.Background(), replacedMasterMachineName, metav1.GetOptions{})
-				// o.Expect(err).NotTo(o.HaveOccurred())
 
-				// add controller reference
-				// replacedMasterMachineCopy := replacedMasterMachine.DeepCopy()
-				// ownerReferences := getOwnerReferenceForMasterMachine(replacedMasterMachineCopy)
-				// replacedMasterMachineCopy.SetOwnerReferences(ownerReferences)
-				// _, err = ms.Update(context.Background(), replacedMasterMachineCopy, metav1.UpdateOptions{})
-				// o.Expect(err).NotTo(o.HaveOccurred())
-
-				targets := []*corev1.Node{ /* replacedMaster , */ replacedWorker}
-				targetMachineNames := []string{ /* replacedMasterMachineName, */ replacedWorkerMachineName}
+				targets := []*corev1.Node{replacedWorker}
+				targetMachineNames := []string{replacedWorkerMachineName}
+				targetNodeNames := sets.NewString()
+				for _, target := range targets {
+					targetNodeNames.Insert(target.Name)
+				}
 
 				// we use a hard shutdown to simulate a poweroff
 				for _, target := range targets {
 					framework.Logf("Forcing shutdown of node %s", target.Name)
-					_, err = ssh("sudo -i systemctl poweroff --force --force", target)
+					ch := make(chan struct{})
+					go func(node *corev1.Node) {
+						defer close(ch)
+						if _, err := ssh("sudo -i systemctl poweroff --force --force", node); err != nil {
```

Comment:
`err` will always be non-nil on successful execution of this command so I'm not sure there's any value in checking it.. In #25707 I've added a new `shutdownNode` function that calls e2essh.SSH directly so that the output can be checked the only accurate indication that shutdown was initiated - the string `Powering off` appearing in stderr.

Reply:
I don't know that there's even a guarantee that powering off would get returned on all clouds.  it was not on GCP at the current time.  So I think even that isn't sufficient.

Reply:
How would it not be returned? And why would it be a function of a particular cloud? - afaict it's RHCOS that's sending `Powering off` to the ssh connection before initiating termination.

Reply:
It's a function of how the load balancers work.  GCP often leaves dangling connections, and since go SSH is limited (can't do idle timeouts correctly) you can't assume you'll ever get a packet (so the SSH hangs forever).  GCP cuts the proxy connection before you get the "powering off" packet, unlike AWS.